test/parallel/test-assert-if-error.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+require('../common');
+const assert = require('assert').strict;
+/* eslint-disable no-restricted-properties */
+
+const err = new Error('test error');
+const msg = err.message;
+
+assert.throws(
+  () => { assert.ifError(err); },
+  {
+    message: 'ifError got unwanted exception: test error',
+    actual: err,
+    expected: null,
+    operator: 'ifError'
+  }
+);
+assert.equal(err.message, msg);
+
+assert.doesNotThrow(() => { assert.ifError(null); });
+assert.doesNotThrow(() => { assert.ifError(); });
+assert.doesNotThrow(() => { assert.ifError(undefined); });
+assert.doesNotThrow(() => { assert.ifError(false); });
```

Comment:
What do others think about actually throwing in such a case? I personally think it would be best to only accept `null` and `undefined`.

Reply:
I agree, it's already semver-major so let's do it. Can you also update the PR description with this change?

Reply:
Done.